price-feeder/oracle/oracle.go

```diff
@@ -163,6 +168,20 @@ func (o *Oracle) GetPrices() map[string]sdk.Dec {
 	return prices
 }
 
+// GetTvwapPrices returns the tvwapsByProvider map using a read lock
+func (o *Oracle) GetTvwapPrices() map[provider.Name]map[string]sdk.Dec {
+	o.tvwapsMutex.RLock()
+	defer o.tvwapsMutex.RUnlock()
+	return o.tvwapsByProvider
+}
+
+// GetVwapPrices returns the vwapsByProvider map using a read lock
+func (o *Oracle) GetVwapPrices() map[provider.Name]map[string]sdk.Dec {
+	o.vwapsMutex.RLock()
+	defer o.vwapsMutex.RUnlock()
+	return o.vwapsByProvider
+}
```

Comment:
What do you think about creating a structure to handle mutex for `map[provider.Name]map[string]sdk.Dec` struct?

Reply:
hm, good question. I'm not super against it but my thinking was that they are already in the oracle struct and if I did that we would probably want to do the same thing with pricesMutex and prices. I'll make some changes and see how it looks.

Reply:
https://github.com/umee-network/umee/pull/1445/commits/61664c8b6373a2f01cc65bb4584e8ca4e820eb26

Reply:
I was trying to suggest something like this #1465